packages/expo-cli/src/commands/upgrade.ts

```diff
@@ -257,6 +257,12 @@ async function upgradeAsync(requestedSdkVersion: string | null, options: Options
     await packageManager.addAsync(...dependenciesAsStringArray);
   }
 
+  // Clear metro bundler cache
+  log.addNewLineIfNone();
+  log(chalk.bold.underline('Clearing your cache to remove old references...'));
+  await Project.startReactNativeServerAsync(projectRoot, { reset: true, nonPersistent: true });
```

Comment:
@fson - `nonPersistent` is gone now, right?

Reply:
I think it's gone, for any SDK 33+, but [still active for people who don't like upgrades](https://github.com/expo/expo-cli/blob/master/packages/xdl/src/Project.ts#L1728) 😄

Reply:
what does the user see when this runs?

Reply:
<details>
<summary>Screenshot before/during dependency logging</summary>
<img src="https://user-images.githubusercontent.com/1203991/68518248-6bc4f100-028b-11ea-94a1-5c205617e152.png" />
</details>

<details>
<summary>Screenshot after dependency logging (including this statement)</summary>
<img src="https://user-images.githubusercontent.com/1203991/68518205-31f3ea80-028b-11ea-8f5b-af5c8a861957.png" />
</details>

<details>
<summary>The full log (tm)</summary>
<pre>
/mnt/projects/peakfijn/bksy/app develop* cedric@vscode 
❯ expo upgrade 
Warning! Your git working tree is dirty. 
It's recommended to commit all your changes before proceeding, 
so you can revert the changes made by this command if necessary. 
 
? Would you like to proceed? Yes 
 
? You are currently using SDK 33.0.0. Would you like to update to the latest version, 35.0.0? Yes 
 
Installing the expo package... 
 
> yarn add expo@^35.0.0 
yarn add v1.19.1 
[1/4] Resolving packages... 
[2/4] Fetching packages... 
info fsevents@1.2.9: The platform "linux" is incompatible with this module. 
info "fsevents@1.2.9" is an optional dependency and failed compatibility check. Excluding it from installation. 
[3/4] Linking dependencies... 
[4/4] Building fresh packages... 
 
success Saved lockfile. 
success Saved 22 new dependencies. 
info Direct dependencies 
└─ expo@35.0.1 
info All dependencies 
├─ @unimodules/core@4.0.0 
├─ @unimodules/react-native-adapter@4.0.0 
├─ babel-preset-expo@7.1.0 
├─ expo-app-loader-provider@7.0.0 
├─ expo-file-system@7.0.0 
├─ expo-keep-awake@7.0.0 
├─ expo-location@7.0.0 
├─ expo-permissions@7.0.0 
├─ expo-sqlite@7.0.0 
├─ expo-web-browser@7.0.1 
├─ expo@35.0.1 
├─ react-native-branch@3.0.1 
├─ unimodules-barcode-scanner-interface@4.0.0 
├─ unimodules-camera-interface@4.0.0 
├─ unimodules-constants-interface@4.0.0 
├─ unimodules-face-detector-interface@4.0.0 
├─ unimodules-file-system-interface@4.0.0 
├─ unimodules-font-interface@4.0.0 
├─ unimodules-image-loader-interface@4.0.0 
├─ unimodules-permissions-interface@4.0.0 
├─ unimodules-sensors-interface@4.0.0 
└─ unimodules-task-manager-interface@4.0.0 
Done in 10.80s. 
 
Updating sdkVersion in app.json... 
Updating packages to compatible versions (where known)... 
 
> yarn add --dev jest-expo@^35.0.0 
yarn add v1.19.1 
[1/4] Resolving packages... 
[2/4] Fetching packages... 
info fsevents@1.2.9: The platform "linux" is incompatible with this module. 
info "fsevents@1.2.9" is an optional dependency and failed compatibility check. Excluding it from installation. 
[3/4] Linking dependencies... 
[4/4] Building fresh packages... 
success Saved lockfile. 
success Saved 3 new dependencies. 
info Direct dependencies 
└─ jest-expo@35.0.0 
info All dependencies 
├─ jest-expo@35.0.0 
├─ jest-watch-select-projects@1.0.0 
└─ jest-watch-typeahead@0.4.0 
Done in 6.32s. 
> yarn add @expo/vector-icons@^10.0.0 expo-asset@~7.0.0 expo-constants@~7.0.0 expo-font@~7.0.0 expo-linear-gradient@~7.0.0 react-native@https://github.com/expo/react-native/archive/sdk-35.0.0.tar.gz react@16.8.3 sentry-expo@~2.0.0 
yarn add v1.19.1 
[1/4] Resolving packages... 
[2/4] Fetching packages... 
info fsevents@1.2.9: The platform "linux" is incompatible with this module. 
info "fsevents@1.2.9" is an optional dependency and failed compatibility check. Excluding it from installation. 
[3/4] Linking dependencies... 
 
success Saved lockfile. 
success Saved 87 new dependencies. 
info Direct dependencies 
├─ react-native@0.59.8 
└─ sentry-expo@2.0.1 
info All dependencies 
├─ @sentry/browser@5.7.1 
├─ @sentry/cli@1.49.0 
├─ @sentry/core@5.7.1 
├─ @sentry/integrations@5.7.1 
├─ @sentry/minimal@5.7.1 
├─ @sentry/react-native@1.0.9 
├─ @sentry/wizard@1.0.1 
├─ @snyk/cli-interface@2.2.0 
├─ @snyk/cocoapods-lockfile-parser@3.0.0 
├─ @snyk/composer-lockfile-parser@1.2.0 
├─ @snyk/dep-graph@1.13.1 
├─ @snyk/gemfile@1.2.0 
├─ @snyk/ruby-semver@2.0.4 
├─ @snyk/snyk-cocoapods-plugin@2.0.1 
├─ @types/agent-base@4.2.0 
├─ @types/bunyan@1.8.6 
├─ @types/debug@4.1.5 
├─ @types/js-yaml@3.12.1 
├─ @types/restify@4.3.6 
├─ @types/semver@5.5.0 
├─ @types/xml2js@0.4.3 
├─ @yarnpkg/lockfile@1.1.0 
├─ ast-types@0.13.2 
├─ bl@3.0.0 
├─ cli-spinner@0.2.10 
├─ clone-deep@0.3.0 
├─ data-uri-to-buffer@1.2.0 
├─ degenerator@1.0.4 
├─ dockerfile-ast@0.0.16 
├─ dotnet-deps-parser@4.5.2 
├─ email-validator@2.0.4 
├─ file-uri-to-path@1.0.0 
├─ fs-constants@1.0.0 
├─ ftp@0.3.10 
├─ get-uri@2.0.4 
├─ git-up@4.0.1 
├─ git-url-parse@11.1.2 
├─ jszip@3.2.2 
├─ lazy-cache@0.2.7 
├─ lie@3.3.0 
├─ lodash.assign@4.2.0 
├─ lodash.assignin@4.2.0 
├─ lodash.clone@4.5.0 
├─ lodash.flatten@4.4.0 
├─ mixin-object@2.0.1 
├─ nconf@0.10.0 
├─ netmask@1.0.6 
├─ object-hash@1.3.1 
├─ pac-proxy-agent@3.0.1 
├─ pac-resolver@3.0.0 
├─ pako@1.0.10 
├─ parse-path@4.0.1 
├─ parse-url@5.0.1 
├─ proxy-agent@3.1.1 
├─ raw-body@2.4.1 
├─ react-native@0.59.8 
├─ secure-keys@1.0.0 
├─ sentry-expo@2.0.1 
├─ set-immediate-shim@1.0.1 
├─ shallow-clone@0.1.2 
├─ snyk-config@2.2.3 
├─ snyk-docker-plugin@1.33.1 
├─ snyk-go-parser@1.3.1 
├─ snyk-go-plugin@1.11.1 
├─ snyk-gradle-plugin@3.2.0 
├─ snyk-module@1.9.1 
├─ snyk-mvn-plugin@2.4.0 
├─ snyk-nodejs-lockfile-parser@1.16.1 
├─ snyk-nuget-plugin@1.13.1 
├─ snyk-paket-parser@1.5.0 
├─ snyk-php-plugin@1.7.0 
├─ snyk-policy@1.13.5 
├─ snyk-python-plugin@1.13.3 
├─ snyk-resolve-deps@4.4.0 
├─ snyk-resolve@1.0.1 
├─ snyk-sbt-plugin@2.8.0 
├─ snyk-try-require@1.3.1 
├─ snyk@1.244.0 
├─ tar-stream@2.1.0 
├─ tempfile@2.0.0 
├─ thunkify@2.1.2 
├─ toml@3.0.0 
├─ tree-kill@1.2.1 
├─ vscode-languageserver-types@3.14.0 
├─ window-size@0.1.4 
├─ xml2js@0.4.22 
└─ xregexp@2.0.0 
Done in 18.80s. 
 
Clearing the packager cache... 
 
Automated upgrade steps complete. 
...but this doesn't mean everything is done yet! 
 
The following packages were updated: 
@expo/vector-icons, expo-asset, expo-constants, expo-font, expo-linear-gradient, jest-expo, react-native, react, sentry-expo, expo 
 
The following packages were not updated. You should check the READMEs for those repositories to determine what version is compatible with your new set of packages: 
@expo/react-native-action-sheet, axios, color, expo-analytics, expo-mixpanel-analytics, lodash, moment, native-base, prop-types, query-string, react-native-google-places-autocomplete, react-native-modal, react-native-star-rating, react-navigation, react-redux, redux, redux-devtools-extension, redux-persist, redux-promise-middleware, redux-thunk, uuid, @peakfijn/config-commitizen, @peakfijn/config-commitlint, @peakfijn/config-eslint-expo, @peakfijn/config-release-expo, babel-plugin-inline-dotenv, dotenv, dotenv-expand, eslint, jest-junit, moxios, react-test-renderer, redux-logger, redux-mock-store 
 
Please refer to the release notes for information on any further required steps to update and information about breaking changes: 
https://blog.expo.io/expo-sdk-35-is-now-available-beee0dfafbf4 
 
You should also look at the breaking changes from a release that you skipped: 
- https://blog.expo.io/expo-sdk-34-is-now-available-4f7825239319 
 
</pre>
</details>